unleash-yggdrasil/src/strategy_parsing.rs

```diff
@@ -327,6 +330,30 @@ fn rollout_constraint(mut node: Pairs<Rule>) -> RuleFragment {
     })
 }
 
+fn get_hostname() -> Result<String, SdkError> {
+    //This is primarily for testing purposes
+    if let Ok(hostname_env) = env::var("HOSTNAME") {
+        return Ok(hostname_env);
+    }
+
+    hostname::get()
+        .map_err(|_| SdkError::StrategyEvaluationError)
+        .and_then(|os_str| {
+            os_str
+                .into_string()
+                .map_err(|_| SdkError::StrategyEvaluationError)
+        })
+}
+
+fn hostname_constraint(inverted: bool, mut node: Pairs<Rule>) -> RuleFragment {
+    let target_hostnames = harvest_set(node.next().unwrap().into_inner());
```

Comment:
This is the magic referenced earlier - we only have to read a single node rather than all 3 nodes since the first two nodes are non emitting. They tell us nothing so they're discarded in the grammar rather than the compiler